src/modules/shlvl.rs

```diff
@@ -219,4 +227,29 @@ mod tests {
 
         assert_eq!(expected, actual);
     }
+
+    #[test]
+    fn repeat_offset() {
+        fn get_actual(shlvl: usize, repeat_offset: usize, threshold: usize) -> Option<String> {
+            ModuleRenderer::new("shlvl")
+                .config(toml::toml! {
+                    [shlvl]
+                    format = "[$symbol]($style)"
+                    symbol = "~"
+                    repeat = true
+                    repeat_offset = repeat_offset
+                    disabled = false
+                    threshold = threshold
+                })
+                .env(SHLVL_ENV_VAR, format!("{}", shlvl))
+                .collect()
+        }
+
+        assert_eq!(get_actual(2, 0, 0), Some(format!("{}", style().paint("~~"))));
+        assert_eq!(get_actual(2, 1, 0), Some(format!("{}", style().paint("~"))));
+        assert_eq!(get_actual(2, 2, 0), None); // offset same as shlvl; hide
+        assert_eq!(get_actual(2, 3, 0), None); // offset larger than shlvl; hide
```

Comment:
I can't think of a reason why anyone would like to keep threshold with repeat_offset settings, but I ensured everything works just for completeness.